src/Link/ApiLinkMacro.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Tomaj\NetteApi\Link;
+
+use Efabrica\Cms\Core\Macro\PageLink;
+use Latte\Compiler;
+use Latte\MacroNode;
+use Latte\Macros\MacroSet;
+use Latte\PhpWriter;
+use Tomaj\NetteApi\EndpointIdentifier;
+
+/**
+ * Usage in latte:
+ * {apiLink $method, $version, $package, $apiAction, ['title' => 'My title', 'data-foo' => 'bar']}
+ */
+class ApiLinkMacro extends MacroSet
+{
+    public static function install(Compiler $compiler)
+    {
+        $macroSet = new static($compiler);
+        $macroSet->addMacro('apiLink', [self::class, 'start']);
+    }
+
+    public static function start(MacroNode $node, PhpWriter $writer)
+    {
+        $args = array_map('trim', explode(',', $node->args, 4));
+        $arguments = [
+            'method' => $args[0] ?? 'null',
+            'version' => $args[1] ?? 'null',
+            'package' => $args[2] ?? 'null',
+            'action' => $args[3] ?? 'null',
+            'params' => $args[4] ?? '[]',
+        ];
+
+        return $writer->write('echo \Tomaj\NetteApi\Link\ApiLinkMacro::createLink($_presenter->context->getByType("' .
+            ApiLink::class . '"), ' .
+            $arguments['method']  . ', ' .
+            $arguments['version']  . ', ' .
+            $arguments['package'] . ', ' .
+            $arguments['action'] . ', ' .
+            $arguments['params'] . ')');
+    }
+
+    /**
+     * @internal
+     */
+    public static function createLink(ApiLink $apiLink, $method, $version, $package, $apiAction = '', $params = [])
```

Comment:
toto sa mi uplne nepaci ze je public, nemozes to napisat cele rovno do toho writera?

Reply:
to ais nepojde skrz ten $_presenter->context->getByType hack